### PR TITLE
Fix SLE12 packages using autotools, re-establish compatibility with OpenCV 3

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/68684#note-9

---

I've also updated the package link for `perl-Test-Warnings`: `osc linkpac openSUSE:Factory perl-Test-Warnings devel:openQA:SLE-12`

I've tested `osc build SLE_12_SP4 x86_64 os-autoinst.spec` and `osc build openSUSE_Tumbleweed x86_64 os-autoinst.spec` locally. (SLE_12_SP4 is the oldest distro we still build packages for in our devel project.)

### CODE REVIEW

File: dist/rpm/os-autoinst.spec [hunk 1]
```diff
@@ -46,14 +50,24 @@ Source0:        %{name}-%{version}.tar.xz
...
+%define test_base_requires %main_requires perl(Benchmark) perl(Devel::Cover) perl(FindBin) perl(Pod::Coverage) perl(Test::Exception) perl(Test::Fatal) perl(Test::Mock::Time) perl(Test::MockModule) perl(Test::MockObject) perl(Test::Mojo) perl(Test::More) perl(Test::Output) perl(Test::Pod) perl(Test::Strict) perl(Test::Warnings) >= 0.029 procps python3-setuptools qemu qemu-tools qemu-x86
+# The following line is generated from dependencies.yaml
+%define test_legacy_requires %build_legacy_requires %test_base_requires
+# The following line is generated from dependencies.yaml
+%define test_requires %build_requires %spellcheck_requires %test_base_requires perl(YAML::PP) python3-yamllint
 # The following line is generated from dependencies.yaml
 %define devel_requires %test_requires perl(Devel::Cover) perl(Devel::Cover::Report::Codecov) perl(Perl::Tidy)
+%if 0%{?suse_version} == 1315
```
Comment (okurz) [inline dist/rpm/os-autoinst.spec:66]: where did you get this exact-match check from? I guess it should be `sle_version` according to https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto

File: t/27-make-update-deps.t [hunk 2]
```diff
@@ -16,12 +16,18 @@
...
 use FindBin '$Bin';
 
 if (not -e "$Bin/../.git") {
-    pass("Skipping all tests, not in a git repository");
+    pass('Skipping all tests, not in a git repository');
+    done_testing;
+    exit;
+}
+if (!can_load(modules => {'YAML::PP' => undef})) {
+    pass('Skipping all tests, YAML::PP is not installed');
```
Comment (okurz) [inline t/27-make-update-deps.t:30]: Not sure, why should it be a good idea to pretend that everything was fine here?

File: dist/rpm/os-autoinst.spec (hunk 1 above)
Comment (Martchus) [reply dist/rpm/os-autoinst.spec:66]: I've got it from https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto

File: t/27-make-update-deps.t (hunk 2 above)
Comment (Martchus) [reply t/27-make-update-deps.t:30]: Yes, I better just remove the file.

**Pull request merged**